pkg/metrics/metrics.go

```diff
@@ -45,72 +48,101 @@ const (
 // is a Client with optOut = true.
 //
 // It is initialized once when called the first time.
-var noopWriterOnce = sync.OnceValue[*Client](func() *Client {
+var noopWriterOnce = sync.OnceValue(func() *Client {
```

Comment:
Why remove the generic?

Does the compiler implicitly know the value so its redundant?

Reply:
Interesting, didn't realize that was a feature, assumed you would get an `any` back and need to cast.